### PR TITLE
Merge envoy-wasm for v1.13.0.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Comment by kyessenov** [comment]
There's a crash in WasmHandle destructor.

**Comment by jplevyak** [comment]
There are a number of changes in the underlying envoy repository that I am not familiar with.
Suggestions appreciated.

**Comment by kyessenov** [comment]
It looks like a Wasm specific crash:
```
[2020-01-24 04:55:22.017][52058][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #9: Envoy::Extensions::Common::Wasm::Context::onDone() [0x558b75aab736]
[2020-01-24 04:55:22.026][52152][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #8: std::__1::function<>::operator()() [0x558b75ab612a]
[2020-01-24 04:55:22.060][52058][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #10: Envoy::Extensions::Common::Wasm::Wasm::shutdown() [0x558b75af2639]
[2020-01-24 04:55:22.068][52152][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #9: Envoy::Extensions::Common::Wasm::Context::onConfigure() [0x558b75aa8e73]
[2020-01-24 04:55:22.101][52058][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #11: Envoy::Extensions::Common::Wasm::WasmHandle::~WasmHandle()::{lambda()#1}::operator()() [0x558b75b1247d]
[2020-01-24 04:55:22.109][52152][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #10: Envoy::Extensions::Common::Wasm::Wasm::configure() [0x558b75af1c03]
[2020-01-24 04:55:22.142][52058][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #12: std::__1::__invoke<>() [0x558b75b1243d]
[2020-01-24 04:55:22.149][52152][critical][backtrace] [bazel-out/k8-fastbuild/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:75] #11: Envoy::Extensions::Common::Wasm::createThreadLocalWasm() [0x558b75af3788]
```

**Comment by kyessenov** [comment]
You also need to re-generate Wasm files here I think.

File: WORKSPACE [hunk 1]
```diff
@@ -37,10 +37,10 @@ bind(
 # 1. Determine SHA256 `wget https://github.com/envoyproxy/envoy-wasm/archive/$COMMIT.tar.gz && sha256sum $COMMIT.tar.gz`
 # 2. Update .bazelversion, envoy.bazelrc and .bazelrc if needed.
 #
-# envoy-wasm commit date: 12/19/19
-ENVOY_SHA = "456bf9aef22d7b4df4916fa49c5dad2f5dbf0f0f"
+# envoy-wasm commit date: Jan 22 2020
```
Comment (douglas-reid) [inline WORKSPACE]: nit: Jan 23, 2020

Comment (jplevyak) [reply WORKSPACE]: Done.

File: WORKSPACE [hunk 2]
```diff
@@ -37,10 +37,10 @@ bind(
 # 1. Determine SHA256 `wget https://github.com/envoyproxy/envoy-wasm/archive/$COMMIT.tar.gz && sha256sum $COMMIT.tar.gz`
 # 2. Update .bazelversion, envoy.bazelrc and .bazelrc if needed.
 #
-# envoy-wasm commit date: 12/19/19
```
Comment (PiotrSikora) [inline WORKSPACE]: Per the text above, you still need to update `.bazelversion` to `2.0.0`.

**Comment by PiotrSikora** [comment]
I kicked off the build on my latop for the macOS, to make sure we don't release from a broken thing.

**Comment by kyessenov** [comment]
@PiotrSikora @jplevyak I'd like to get https://github.com/envoyproxy/envoy-wasm/pull/345 in. I can send a follow-up SHA bump or you can include it here.

**Comment by kyessenov** [comment]
Btw I assume the crash was due to ABI change. Should I file an issue to harden host once more, since it's likely to happen someone is going to run mis-matched Wasm file anyways?

**Comment by jplevyak** [comment]
Kuat, do we need that for the release?  It isn't currently used AFAICT

**Comment by jplevyak** [comment]
We should probably just get this one in.

**Comment by kyessenov** [comment]
I think so. Why not? It's fairly self-contained from the rest. I can make a plugin PR to use it quickly, since I was waiting for it to make through. It would be nice to test out de-coupling plugin release from envoy release actually.

**Comment by PiotrSikora** [comment]
Yes, sorry, still waiting for one of the macOS builds to finish (either on CircleCI or my laptop, whichever is first).

**Pull request merged**